gateway/api_loader.go

```diff
@@ -918,12 +937,14 @@ func (gw *Gateway) loadApps(specs []*APISpec) {
 
 	gw.DefaultProxyMux.swap(muxer, gw)
 
-	// Swap in the new register
 	gw.apisMu.Lock()
 
-	// release current specs resources before overwriting map
-	for _, curSpec := range gw.apisByID {
-		curSpec.Release()
+	for _, spec := range specs {
+		for _, curSpec := range gw.apisByID {
```

Comment:
The key of apisByID map is the APIID. This is an O(N) which could be O(1)?

Reply:
using getAPISpec now, resolved